crates/rattler_conda_types/src/channel/mod.rs

```diff
@@ -28,6 +28,10 @@ pub struct ChannelConfig {
     ///
     /// The default value is: <https://conda.anaconda.org>
     pub channel_alias: Url,
+
+    /// For local channels, the root directory from which to resolve relative paths.
+    /// Most of the time you would initialize this with the current working directory.
+    pub root_dir: PathBuf,
```

Comment:
Maybe we can make this Option to make it easier to construct this type. If its None then we use the current directory and otherwise we use the explicitly set one. WDYT?

Reply:
Using current directory is kinda annoying because it can fail though. That's why I liked to insert the type from outside.

Reply:
Makes sense but now the Default impl can fail without a way to recover. Maybe we should temove that too in favor of a failable associated function?